src/main.rs

```diff
@@ -298,4 +322,39 @@ mod tests {
             out[0]
         );
     }
+
+    #[test]
+    fn extract_volume_metrics_with_empty_results() {
+        let mut out = Vec::new();
+        extract_volume_metrics(&json!([]), "volume", &mut out);
+        assert_eq!(
+            AppsignalMetric::new(
+                "volume_fs_available_bytes",
+                HashMap::from([("pod".to_string(), "volume".to_string())]),
+                &json!(0.0)
+            ),
+            out[0]
+        );
+    }
+
+    #[test]
+    fn extract_volume_metrics_with_results() {
+        let mut out = Vec::new();
+        extract_volume_metrics(
+            &json!({
+              "availableBytes": 232839439
+            }),
+            "volume",
+            &mut out,
+        );
+
+        assert_eq!(
+            AppsignalMetric::new(
+                "volume_available_bytes",
+                HashMap::from([("volume".to_string(), "volume".to_string())]),
```

Comment:
Does this metric only have the volume name as a tag?
That could easily clash with metrics from different pods with the same volume names. Can we add the pod or something else that is identifiable?

Reply:
My understanding, which might be wrong, was that volume names are unique ([object names are unique](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/), and volumes are objects? _maybe?_) but that might be wrong (there's a section regarding the possibility of [ephemeral volume name collision](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#persistentvolumeclaim-naming)?)

I'll see about adding the pod as well, though this creates the situation where if the same persistent volume is mounted in two different pods, we report the same metric twice. Still preferable to the current implementation, of course.

Reply:
It's not clear to me what such an ID looks like. I know that volumes can have a name in the Kubernetes config, whatever the developer enters. That's not the name we get back from the API?

Are they namespace per pod, or something like UUIDs? (If they are UUIDs, that makes them less readable.)

> I'll see about adding the pod as well, though this creates the situation where if the same persistent volume is mounted in two different pods, we report the same metric twice. Still preferable to the current implementation potentially sending two different metrics with the same tags, of course.

If the same volume is reported twice, it would be better if one is removed, deduplicated.
If the volume name is unique across the entire cluster, then no need to add the pod.

Reply:
I'll try and find out, yes. I suspect the answer might be "it depends", possibly on the kind of volume it is.

Reply:
Confirmed that different pods (e.g. in different deployments) can have volumes with the same name. Will add the pod name to the tag set.